### PR TITLE
List of cryptocurrencies

### PR DESCRIPTION
**Pull request by AbrahamAN**
This pull request adds a new functionality to the project: a list of all cryptocurrencies. The list is in a new file called CryptoCoinItem.tsx and contains information about the name, symbol, and code of each cryptocurrency.

Demostrative Video
https://www.loom.com/share/1765d39ffaea426a96c4f6ac28fb9dce

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import MyCryptosScreen from './screens/MyCryptosScreen';
+import {NavigationContainer} from '@react-navigation/native';
+import {createNativeStackNavigator} from '@react-navigation/native-stack';
+import {AddCryptoScreen} from './screens/AddCryptoScreen';
+
+const Stack = createNativeStackNavigator();
+
+const App = () => {
```
Comment (ThomasDevlabs) [inline src/App.tsx:9]: use implicit return 

const App = () => (<,,,,)

Comment (tomasdematos) [inline src/App.tsx]: const App = () => (<..../>)

File: src/components/AddButton.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import {Text} from 'react-native';
+import styled from 'styled-components/native';
+
+const Container = styled.View`
```
Comment (tomasdematos) [inline src/components/AddButton.tsx]: styles should be stored in a separate file, the most common way to do this is with this format:

```
src/
    Components/
      Container/
        index.tsx <---- this file
        styles.ts <---- move the styles here
```

File: src/components/AddButton.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  padding: 10px;
+  align-items: center;
+`;
+
+interface AddButtonProps {
+  onPress: () => void;
+}
+
+const AddButton = ({onPress}: AddButtonProps) => {
+  return (
```
Comment (tomasdematos) [inline src/components/AddButton.tsx]: use implicit return, if there is nothing before the return

File: src/components/AddButton.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    <Container>
+      <Text onPress={onPress} style={{color: 'grey', fontSize: 18}}>
```
Comment (tomasdematos) [inline src/components/AddButton.tsx]: don't use inline styles

File: src/components/CryptoCoinItem.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+import Avatar from './Avatar';
+import {CryptoApiResponse, CryptoCoin} from './types';
+
+export const CryptoCoinItem = () => {
+  const [isLoading, setLoading] = useState<boolean>(true);
+  const [crypto, setCrypto] = useState<CryptoCoin[]>([]);
+
+  const getCrypto = async () => {
+    try {
+      const response = await fetch('https://data.messari.io/api/v1/assets');
```
Comment (tomasdematos) [inline src/components/CryptoCoinItem.tsx]: the conecction with the api is outside the scope, it should be implemented in the correct PR

File: src/components/CryptoCoinItem.tsx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+  }, []);
+
+  const Container = styled.View({
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    padding: 10,
+    margin: 25,
+    borderBottomWidth: 1,
+    borderBottomColor: 'grey',
```
Comment (tomasdematos) [inline src/components/CryptoCoinItem.tsx]: the colors should be stored in a theme file: 

src/utils/theme.tsx

```
export const theme = {
  colors: {
    white: "#ffffff",
    .....
   }
}
```

File: src/screens/AddCryptoScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import {Button, View} from 'react-native';
+import styled from 'styled-components';
+import Header from '../components/Header';
+
+const B = styled.Button`
```
Comment (tomasdematos) [inline src/screens/AddCryptoScreen.tsx:6]: use more descriptive names

**Pull request closed without merging**